scripts/common/_common.sh

```diff
@@ -0,0 +1,40 @@
+#
+# Copyright (c) .NET Foundation and contributors. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+#
+
+COMMONSOURCE="${BASH_SOURCE[0]}"
+while [ -h "$SOURCE" ]; do # resolve $SOURCE until the file is no longer a symlink
+  COMMONDIR="$( cd -P "$( dirname "$COMMONSOURCE" )" && pwd )"
+  COMMONSOURCE="$(readlink "$COMMONSOURCE")"
+  [[ $COMMONSOURCE != /* ]] && COMMONSOURCE="$COMMONDIR/$COMMONSOURCE" # if $SOURCE was a relative symlink, we need to resolve it relative to the path where the symlink file was located
+done
+DIR="$( cd -P "$( dirname "$COMMONSOURCE" )" && pwd )"
```

Comment:
This should be COMMONDIR

When this script is sourced, all vars (not only those exported) will be brought into the calling script. Using COMMONDIR here avoids a collision. Without it you may be getting some weird path errors.